libs/libarchfpga/src/arch_util.cpp

```diff
@@ -613,10 +610,9 @@ void ProcessLutClass(t_pb_type* lut_pb_type) {
 
     lut_pb_type->modes[0].interconnect[0].parent_mode_index = 0;
     lut_pb_type->modes[0].interconnect[0].parent_mode = &lut_pb_type->modes[0];
-    lut_pb_type->modes[0].interconnect[0].interconnect_power = (t_interconnect_power*)vtr::calloc(1, sizeof(t_interconnect_power));
+    lut_pb_type->modes[0].interconnect[0].interconnect_power = new t_interconnect_power();
 
-    lut_pb_type->modes[0].interconnect[0].annotations = (t_pin_to_pin_annotation*)vtr::calloc(lut_pb_type->num_annotations,
-                                                                                              sizeof(t_pin_to_pin_annotation));
+    lut_pb_type->modes[0].interconnect[0].annotations = new t_pin_to_pin_annotation[lut_pb_type->num_annotations]();
```

Comment:
calloc -> new: unsafe unless you have a default constructor for annotations (if you use the vector of pair<int,string> suggested by @soheilshahrouz you will have a default constructor since all standard library classes have one.